torch/csrc/utils/python_arg_parser.h

```diff
@@ -658,6 +658,9 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(
         } catch (std::exception& e) {
           throw_intlist_exception(this, i, obj, idx, e);
         }
+      } else if (torch::is_symint(py::handle(obj))) {
+        res[idx] = py::cast<c10::SymInt>(py::handle(obj))
+                       .guard_int(__FILE__, __LINE__);
```

Comment:
quite curious, why don't we guard in symintlist case?

https://github.com/pytorch/pytorch/blob/8162f4170b8c7df3bab503103544e1d7cbf8b052/torch/csrc/utils/python_arg_parser.h#L534

Reply:
Is this for the so-called unbacked symint case? :thinking:

Reply:
Oh right, this guard is necessary to extract the backed values, which the downstream aten op requires?

Reply:
You don't need to guard for symintlist because we can just return the SymInts directly, to track the symbolic flow. That's the point of having SymInts.

The guard here is for operators that don't advertise symint support. Stuff like dim=[1, 2]. We never handle this symbolically, always specialize.

Reply:
So essentially, they flow through the program as a DAG, and can terminate at leaves where they are specialized. I suppose there are also scenarios where they are outputs of the program and are never specialized.

Reply:
They form a DAG when they're not specialized. If we specialize, we don't actually need to model its compute because we just inline constants everywhere.